src/widget/form/addfriendform.cpp

```diff
@@ -143,6 +143,18 @@ Ignore the proxy and connect to the Internet directly?"), QMessageBox::Yes|QMess
     }
 }
 
+void AddFriendForm::onIdChanged(const QString &id)
+{
+    QString tId = id.trimmed();
+    QRegularExpression dnsIdExpression("^\\S+@\\S+$");
+    bool isValidId = ToxId::isToxId(tId) || tId.contains(dnsIdExpression);
+
+    toxId.setStyleSheet(isValidId ? QStringLiteral("") : QStringLiteral("QLineEdit { background-color: #FFC1C1; }"));
+    toxId.setToolTip(isValidId ? QStringLiteral("") : tr("Invalid Tox ID format. It must have length of 76 hexadecimal characters or similar name@domain.com."));
```

Comment:
I don't think that setting tooltip is enough - e.g. I didn't even think to hoover over input field to get some more info, and when things "didn't work", only then I've looked at code, to find out that tooltip is used.

What needs to be done in order to clearly communicate to user that it's not an ID, is to show a warning dialog window.